packages/framework/aqueduct/src/data-objects/pureDataObject.ts

```diff
@@ -24,13 +24,25 @@ import { handleFromLegacyUri } from "@fluidframework/request-handler";
 import { serviceRoutePathRoot } from "../container-services";
 import { defaultFluidObjectRequestHandler } from "../request-handlers";
 
-// eslint-disable-next-line @typescript-eslint/ban-types
-export interface IDataObjectProps<O = object, S = undefined> {
+export interface DataObjectTypes {
+    O?: FluidObject,
+    S?: any,
+    E?: IEvent
+}
+
+export type Default<T extends DataObjectTypes> = {
```

Comment:
this guy then lets us pick types off of the user provided `DataObjectTypes` and fill in defaults for the types they didn't set

Reply:
this needs a better name

Reply:
Maybe `UsedDataObjectTypes`?  Since it describes the types that will ultimately be used after considering both the configuration set by the customer as well as the defaults?

Also would be good to add a comment here for those of us less-fluent in TS :)